dpctl/tensor/_ctors.py

```diff
@@ -0,0 +1,377 @@
+        # new USM allocation
+        res = dpt.usm_ndarray(
+            usm_ndary.shape,
+            dtype=dtype,
+            buffer=usm_type,
+            order="C",
+            buffer_ctor_kwargs={"queue": copy_q},
+        )
+        original_strides = usm_ndary.strides
+        ind = sorted(
+            range(usm_ndary.ndim),
+            key=lambda i: abs(original_strides[i]),
+            reverse=True,
+        )
+        new_strides = tuple(res.strides[ind[i]] for i in ind)
+        # reuse previously made USM allocation
+        res = dpt.usm_ndarray(
+            usm_ndary.shape,
+            dtype=res.dtype,
+            buffer=res.usm_data,
+            strides=new_strides,
+        )
+    else:
+        res = dpt.usm_ndarray(
+            usm_ndary.shape,
+            dtype=dtype,
+            buffer=usm_type,
+            order=order,
+            buffer_ctor_kwargs={"queue": copy_q},
+        )
+    # FIXME: call copy_to when implemented
+    res[(slice(None, None, None),) * res.ndim] = usm_ndary
+    return res
+
+
+def _asarray_from_numpy_ndarray(
+    ary, dtype=None, usm_type=None, sycl_queue=None, order="K"
+):
+    if not isinstance(ary, np.ndarray):
+        raise TypeError(f"Expected numpy.ndarray, got {type(ary)}")
+    if usm_type is None:
+        usm_type = "device"
+    if dtype is None:
+        dtype = ary.dtype
+    copy_q = _normalize_queue_device(q=None, d=sycl_queue)
+    f_contig = ary.flags["F"]
+    c_contig = ary.flags["C"]
+    fc_contig = f_contig or c_contig
+    if order == "A":
+        order = "F" if f_contig and not c_contig else "C"
+    if order == "K" and fc_contig:
+        order = "C" if c_contig else "F"
+    if order == "K":
+        # new USM allocation
+        res = dpt.usm_ndarray(
+            ary.shape,
+            dtype=dtype,
+            buffer=usm_type,
+            order="C",
+            buffer_ctor_kwargs={"queue": copy_q},
+        )
+        original_strides = ary.strides
+        ind = sorted(
+            range(ary.ndim),
+            key=lambda i: abs(original_strides[i]),
+            reverse=True,
+        )
+        new_strides = tuple(res.strides[ind[i]] for i in ind)
+        # reuse previously made USM allocation
+        res = dpt.usm_ndarray(
+            res.shape, dtype=res.dtype, buffer=res.usm_data, strides=new_strides
+        )
+    else:
+        res = dpt.usm_ndarray(
+            ary.shape,
+            dtype=dtype,
+            buffer=usm_type,
+            order=order,
+            buffer_ctor_kwargs={"queue": copy_q},
+        )
+    # FIXME: call copy_to when implemented
+    res[(slice(None, None, None),) * res.ndim] = ary
+    return res
+
+
+def asarray(
+    obj,
+    dtype=None,
+    device=None,
+    copy=None,
+    usm_type=None,
+    sycl_queue=None,
+    order="K",
+):
+    """Represents object `obj` as usm_ndarray"""
+    # 1. Check that copy is a valid keyword
+    if copy not in [None, True, False]:
+        raise TypeError(
```

Comment:
Seems `ValueError` is more suitable here.

Reply:
I disagree, it my mind the check is asking if `copy` is of `bool` type, or of `None` type.